serde_str_helpers/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+// Rust language amplification library providing multiple generic trait
+// implementations, type wrappers, derive macros and other language enhancements
+//
+// Written in 2019-2020 by
+//     Martin Habovstiak <martin.habovstiak@gmail.com>
+//     Dr. Maxim Orlovsky <orlovsky@pandoracore.com>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the MIT License
+// along with this software.
+// If not, see <https://opensource.org/licenses/MIT>.
+
+//! This crate contains helpers for using serde with strings.
+//!
+//! Currently there is only a helper for deserializing stringly values more
+//! efficiently by avoiding allocation (and copying) in certain cases.
+
+use std::borrow::Cow;
+use std::ops::Deref;
+
+/// This is a helper for deserializing using `TryFrom` more efficiently.
+///
+/// When using `#[serde(try_from = "String"]` when deserializing a value that
+/// doesn't neet to hold the string (e.g. an integer value) `serde` would
+/// allocate the string even if it doesn't have to.
+///
+/// A naive idea is to use `std::borrow::Cow` to solve it. Sadly, the
+/// implementation of Deserialize for Cow<'de, str> doesn't borrow the string,
+/// so it still allocates needlessly. This helper solves the issue.
+///
+/// Our DeserStrHelper is written such that it borrows the `str` when possible,
+/// avoiding the allocation. It may still need to allocate, for example if
+/// string decoding (unescaping) has to be performed.
+///
+/// ## Example
+///
+/// ```
+/// use serde_derive::Deserialize;
+/// use serde_str_helpers::DeserStrHelper;
+/// use std::convert::TryFrom;
+///
+/// #[derive(Deserialize)]
+/// #[serde(try_from = "DeserStrHelper")]
```

Comment:
From this sample it follows that 
```rust
#[serde(try_from = "BorrowStr")]
```
may be more nice as a naming?

Reply:
Hmm, good question. It looks quite worse on its own though. `DeserBorrowStr`?